chain/chain/src/types.rs

```diff
@@ -407,9 +413,38 @@ mod tests {
             2,
             b1.header.inner.epoch_id.clone(),
             approvals,
-            signer.clone(),
+            &*signer,
         );
         assert!(signer.verify(b2.hash().as_ref(), &b2.header.signature));
         assert_eq!(b2.header.inner.total_weight.to_num(), 3);
     }
 }
+
+/// Verifies that chunk's proofs in the header match the body.
+pub fn validate_chunk_proofs(
+    chunk: &ShardChunk,
+    runtime_adapter: &dyn RuntimeAdapter,
+) -> Result<bool, Error> {
```

Comment:
Result<bool, Error> => bool ?

Reply:
IMO it's better to propagate error -> because that's serialization errors. upstream can decide to print it / consume it.